### PR TITLE
reduce test time by waiting for conditions

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

Here are some small changes to speed up testing and hopefully we can get under the 60s timeout on travis.  These changes made so locally I was able to run the tests in ~8s vs the ~18s it took before.

Most of what I did was change places where we waited for a fixed amount of time to instead wait until the condition was fulfilled.  I also added a bunch of instrumentation around how long different steps take.  Part of the goal of that is to see if there is some specific step that is taking a long time in the travis logs and if there is anything we can do about it.

### CODE REVIEW

**Comment by tylerjw** [comment]
I added that last commit just to test a really long timeout to see if it'll pass with either the default 60s or a longer one.  I don't know if I expect it to pass with the shorter one and I don't really want 5min of CI to be spent testing servo.  We'll see.  I created a PR over on moveit2 just to get CI to run.  I intend for this to be the one where we talk about these changes though.  @AdamPettinger

**Comment by tylerjw** [comment]
Reset it to just 60s.  It seems like this is taking about 14s to pass now :)

**Comment by tylerjw** [comment]
This is now passing the CI tests in 14.3s on travis :)

**Comment by tylerjw** [comment]
It is passing clang-tidy too now... https://github.com/ros-planning/moveit2/pull/253

File: moveit_ros/moveit_servo/src/collision_check.cpp [hunk 1]
```diff
@@ -218,7 +218,7 @@ void CollisionCheck::printCollisionPairs(collision_detection::CollisionResult::C
                                     << contact_map.begin()->first.first << ", " << contact_map.begin()->first.second);
     // Log all other contacts if in debug mode
     RCLCPP_DEBUG_STREAM_THROTTLE(LOGGER, clock, ROS_LOG_THROTTLE_PERIOD, "Objects in collision:");
-    for (auto contact : contact_map)
+    for (const auto& contact : contact_map)
```
Comment (AdamPettinger) [inline moveit_ros/moveit_servo/src/collision_check.cpp:221]: I imagine this change should also be made on ROS1? I can add it to the open PR I have over there where I have added other small changes found from the ROS2 review (https://github.com/ros-planning/moveit/pull/2249).

File: moveit_ros/moveit_servo/test/test_servo_collision.cpp [hunk 2]
```diff
@@ -45,43 +45,49 @@ namespace moveit_servo
...
 
   // Look for DECELERATE_FOR_COLLISION status
   watchForStatus(moveit_servo::StatusCode::DECELERATE_FOR_COLLISION);
 
   // Publish some joint jog commands that will bring us to collision
-  rclcpp::Rate loop_rate(20);
-  for (size_t i = 0; i < 40; ++i)
+  log_time_start = node_->now();
+  size_t iterations = 0;
+  while (!sawTrackedStatus() && iterations++ < TIMEOUT_ITERATIONS)
```
Comment (AdamPettinger) [inline moveit_ros/moveit_servo/test/test_servo_collision.cpp:63]: I like way a lot more than publishing for some time and checking at the end

File: moveit_ros/moveit_servo/test/servo_launch_test_common.hpp [hunk 3]
```diff
@@ -361,6 +360,70 @@ class ServoFixture : public ::testing::Test
     return status_seen_;
   }
 
+  bool start()
```
Comment (AdamPettinger) [inline moveit_ros/moveit_servo/test/servo_launch_test_common.hpp:363]: I like this

**Review by AdamPettinger — APPROVED** [review]
These look awesome, thanks @tylerjw!

File: moveit_ros/moveit_servo/src/collision_check.cpp (hunk 1 above)
Comment (AdamPettinger) [reply moveit_ros/moveit_servo/src/collision_check.cpp:221]: [That line in ROS1](https://github.com/ros-planning/moveit/blob/ab68828e546ac6b9fdecb2c3d38fe07311a863d2/moveit_ros/moveit_servo/src/collision_check.cpp#L221)

**Pull request merged**